sdk/src/main/scala/com/horizen/certificatesubmitter/NonCeasingCertificateSubmitter.scala

```diff
@@ -0,0 +1,293 @@
+              case Some(_) => // do nothing
+              case None =>
+                val referencedWithdrawalEpochNumber = submissionWindowStatus.withdrawalEpochInfo.epoch - 1
+                getMessageToSign(referencedWithdrawalEpochNumber) match {
+                  case Success(messageToSign) =>
+                    signaturesStatus = Some(SignaturesStatus(referencedWithdrawalEpochNumber, messageToSign, ArrayBuffer()))
+
+                    // Try to calculate signatures if signing is enabled
+                    if (certificateSigningEnabled) {
+                      calculateSignatures(messageToSign) match {
+                        case Success(signaturesInfo) =>
+                          signaturesInfo.foreach(sigInfo => {
+                            self ! LocallyGeneratedSignature(sigInfo)
+                          })
+                        case Failure(exception) =>
+                          log.warn(s"Unexpected behavior on SemanticallySuccessfulModifier($block) while calculating signatures.", exception)
+                          signaturesStatus = None // keep signaturesStatus undefined as before
+                      }
+                    }
+                  case Failure(exception) =>
+                    log.warn(s"Unexpected behavior on SemanticallySuccessfulModifier($block) while calculating message to sign.", exception)
+                    signaturesStatus = None // keep signaturesStatus undefined as before
+                }
+            }
+          } else {
+            if (timers.isTimerActive(CertificateGenerationTimer)) {
+              timers.cancel(CertificateGenerationTimer)
+              log.info("Cancel the scheduled Certificate generation due to the Submission Window end")
+              context.system.eventStream.publish(CertificateSubmissionStopped)
+            }
+            signaturesStatus = None
+          }
+
+        case Failure(exception) =>
+          log.warn(s"Unexpected behavior on SemanticallySuccessfulModifier($block) while calculating SubmissionWindowStatus.", exception)
+      }
+  }
+
+  // Take withdrawal epoch info for block from the History.
+  // Note: We can't rely on `State.getWithdrawalEpochInfo`, because it shows the tip info,
+  // but the older block may being applied at the moment.
+  private[certificatesubmitter] def getSubmissionWindowStatus(block: SidechainBlock): Try[SubmissionWindowStatus] = Try {
+    def getStatus(sidechainNodeView: View): SubmissionWindowStatus = {
+      val withdrawalEpochInfo: WithdrawalEpochInfo = sidechainNodeView.history.blockInfoById(block.id).withdrawalEpochInfo
+      SubmissionWindowStatus(withdrawalEpochInfo, WithdrawalEpochUtils.inSubmitCertificateWindow(withdrawalEpochInfo, params))
+    }
+
+    Await.result(sidechainNodeViewHolderRef ? GetDataFromCurrentView(getStatus), timeoutDuration).asInstanceOf[SubmissionWindowStatus]
+  }
+
+  private[certificatesubmitter] def tryToGenerateCertificate: Receive = {
```

Comment:
any difference between ceasing and non-ceasing cases for proof and cert generation?